deploy/crds/kieapp_crd.yaml

```diff
@@ -206,3 +206,167 @@ spec:
                             secret:
                               type: string
                               description: Secret value for webhook
+            auth:
+              type: object
+              description: Authentication integration configuration
+              oneOf:
+              - required:
+                - sso
+              - required:
+                - ldap
+              properties:
+                sso:
+                  type: object
+                  description: RH-SSO integration configuration
+                  required:
+                  - url
+                  - realm
+                  properties:
+                    url:
+                      type: string
+                      description: RH-SSO URL
+                    realm:
+                      type: string
+                      description: RH-SSO Realm name
+                    adminUser:
+                      type: string
+                      description: RH-SSO Realm Admin Username used to create the Client if it doesn't exist
+                    adminPassword:
+                      type: string
+                      format: password
+                      description: RH-SSO Realm Admin Password used to create the Client
+                    disableSSLCertValidation:
+                      type: boolean
+                      description: RH-SSO Disable SSL Certificate Validation
+                    principalAttribute:
+                      type: string
+                      description: RH-SSO Principal Attribute to use as username
+                    clients:
+                      type: object
+                      description: Client definitions used for creating the RH-SSO clients in the specified Realm
+                      properties:
+                        console:
+                          type: object
+                          description: Console client definition
```

Comment:
If you have console in there, I imagine some of the fields should be required?

Reply:
I'm not sure because having the console client defined, although empty, will have the controller set the hostnameHTTPS. The client name is generated by the scripts and the secret can be generated by the RH-SSO.

Reply:
I take that back. I think it's better to set the client name mandatory.